### PR TITLE
[ci] Add Codecov code coverage for cloud Go files

### PR DESCRIPTION
**Pull request by hcgatewood**
## Summary

Report code coverage for cloud Go tests to Codecov. See https://github.com/magma/magma/pull/4647 for more info on Codecov.

### Workflow overview

- *Generate* coverage
	- Per-subproject (per go.mod)
	- Generate with [go-acc](https://github.com/ory/go-acc), a wrapper around `go coverage` which enables cross-package statistics (e.g. if a test in the state package uses a function from the serde package, record that as a positive coverage of the serde function)
- *Filter* out tools and generated files from coverage report
- *Merge* the per-subproject coverage reports using [gocovmerge](https://github.com/wadey/gocovmerge)
- *Upload* coverage info to Codecov

### Screenshots

Screenshots of [a code coverage upload from this PR](https://codecov.io/gh/magma/magma/tree/09df38e0a36966de80274c60f864c87f7fab5eeb)

Screen Shot 2021-01-29 at 2 04 18 AM
Screen Shot 2021-01-29 at 2 04 53 AM
Screen Shot 2021-01-29 at 2 05 06 AM
Screen Shot 2021-01-29 at 2 05 23 AM

## Test Plan

Manual inspection

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]

**Review by mpgermano — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -490,12 +491,21 @@ jobs:
...
           name: Lint cloud Go code
           command: |
             cd ${MAGMA_ROOT}/orc8r/cloud/docker
             python3 build.py --lint
+      - run:
+          name: Generate test coverage
+          command: |
+            cd ${MAGMA_ROOT}/orc8r/cloud/docker
+            python3 build.py --coverage
+      - run:
```
Comment (tmdzk) [inline .circleci/config.yml]: I think @bbarritt had a CircleCI orb in order to do that.

File: orc8r/cloud/docker/controller/apidocs/v1/swagger.yml [hunk 2]
```diff
@@ -3100,8 +3100,7 @@ paths:
       - $ref: '#/parameters/network_id'
       responses:
         "200":
-          description: List of all MSISDNS in the network, mapped to their subscriber
-            ID
+          description: List of all MSISDNS in the network, mapped to their subscriber ID
```
Comment (tmdzk) [inline orc8r/cloud/docker/controller/apidocs/v1/swagger.yml:3103]: let's try to keep one PR per feature in the future. It's not that I don't like to review your code haha, but it's easier to rollback as we squash before merging to master.

**Review by uri200 — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (hcgatewood) [reply .circleci/config.yml]: The orb you're referencing is a third-party orb, which means we need to explicitly allow third-party orbs

If you want to go ahead and enable that for our project then I can switch over to using the orb 🙂

File: orc8r/cloud/docker/controller/apidocs/v1/swagger.yml (hunk 2 above)
Comment (hcgatewood) [reply orc8r/cloud/docker/controller/apidocs/v1/swagger.yml:3103]: This file is generated. The change looks like it's coming from the go.mod changes, which in turn come from the changed tool dependencies

So it's a necessary part of this PR

File: .circleci/config.yml (hunk 1 above)
Comment (tmdzk) [reply .circleci/config.yml]: I already enabled it!

File: orc8r/cloud/docker/controller/apidocs/v1/swagger.yml (hunk 2 above)
Comment (tmdzk) [reply orc8r/cloud/docker/controller/apidocs/v1/swagger.yml:3103]: oh my bad!

**Review by tmdzk — APPROVED** [review]

**Pull request merged**